.github/workflows/tests.yml

```diff
@@ -7,11 +7,11 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.7, 3.8, 3.9]
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
```

Comment:
Can we keep 3.8?

Reply:
Do we specifically need 3.8? It will be end of life later this year.

Reply:
For anyone on focal, 3.8 is required.  We can remove a few months following 3.8 EOL.